app-backend/database/src/main/scala/com/azavea/rf/database/tables/Shapes.scala

```diff
@@ -0,0 +1,178 @@
+
+  lazy val organizationsFk = foreignKey("shapes_organization_id_fkey", organizationId, Organizations)(r => r.id, onUpdate=ForeignKeyAction.NoAction, onDelete=ForeignKeyAction.NoAction)
+  lazy val createdByUserFK = foreignKey("shapes_created_by_fkey", createdBy, Users)(r => r.id, onUpdate=ForeignKeyAction.NoAction, onDelete=ForeignKeyAction.NoAction)
+  lazy val modifiedByUserFK = foreignKey("shapes_modified_by_fkey", modifiedBy, Users)(r => r.id, onUpdate=ForeignKeyAction.NoAction, onDelete=ForeignKeyAction.NoAction)
+  lazy val ownerUserFK = foreignKey("shapes_owner_fkey", owner, Users)(r => r.id, onUpdate=ForeignKeyAction.NoAction, onDelete=ForeignKeyAction.NoAction)
+}
+
+object Shapes extends TableQuery(tag => new Shapes(tag)) with LazyLogging {
+
+  val tq = TableQuery[Shapes]
+  type TableQuery = Query[Shapes, ShapeModel, Seq]
+
+
+  implicit class withShapesTableQuery[M, U, C[_]](shapes: Shapes.TableQuery) extends
+    ShapeTableQuery[M, U, C](shapes)
+
+  /** List shapes given a page request
+    *
+    * @param offset Int offset of request for pagination
+    * @param limit Int limit of objects per page
+    * @param queryParams ShapeQueryparameters query parameters for request
+    */
+  def listShapes(offset: Int, limit: Int, queryParams: ShapeQueryParameters, user: User) = {
+
+    val dropRecords = limit * offset
+    val filteredShapes = Shapes
+      .filterByUser(queryParams.userParams)
+    val pagedShapes = filteredShapes
+         .drop(dropRecords)
+         .take(limit)
+
+    ListQueryResult[ShapeModel](
+      (pagedShapes.result):DBIO[Seq[ShapeModel]],
+      filteredShapes.length.result
+    )
+  }
+
+  /** Insert a Shape given a create case class with a user
+    *
+    * @param shapeToCreate Shape.Create object to use to create full Shape
+    * @param user               User to create a new Shape with
+    */
+  def insertShape(shapeToCreate: ShapeModel.Create, user: User) = {
+    val shape = shapeToCreate.toShape(user)
+    (Shapes returning Shapes).forceInsert(shape)
+  }
+
+  /** Insert many Shapes given a secreate case class with a user
+    *
+    * @param shapesToCreate Seq[Shape.Create] to use to create full Shape
+    * @param user                User to create a new Shape with
+    */
+  def insertShapes(shapesToCreate: Seq[ShapeModel.Create], user: User)
+                       (implicit database: DB) = {
+    val shapesToInsert = shapesToCreate map { _.toShape(user) }
+    database.db.run {
+      Shapes.forceInsertAll(
+        shapesToInsert
+      )
+    } map { res =>
+      shapesToInsert.map(_.toGeoJSONFeature)
+    }
+  }
+
+
+  /** Given a Shape ID, attempt to retrieve it from the database
+    *
+    * @param shapeId UUID ID of shape to get from database
+    * @param user     Results will be limited to user's organization
+    */
+  def getShape(shapeId: UUID, user: User) =
+    Shapes
+      .filterToSharedOrganizationIfNotInRoot(user)
```

Comment:
I think this should be `filterByUser` maybe too? This matches above for now -- though in the future there will need to be some way to filter by organization as well

Reply:
This applies to I think all the endpoints here

Reply:
👍